Lib/Protocols/IdHL7.pas

```diff
@@ -315,33 +331,39 @@   TIdHL7 = class(TIdBaseComponent)
 
     procedure WaitForConnection(AMaxLength: Integer); // milliseconds
 
+    function ConvertIPtoCardinal(const AStr: AnsiString): Cardinal;
+
     // asynchronous.
-    function AsynchronousSend(AMsg: String): TSendResponse;
+    function AsynchronousSend(AMsg: AnsiString): TSendResponse;
     property OnMessageArrive: TMessageArriveEvent Read FOnMessageArrive Write FOnMessageArrive;
 
     // synchronous
-    function SynchronousSend(AMsg: String; var VReply: String): TSendResponse;
+    function SynchronousSend(AMsg: AnsiString; var VReply: AnsiString): TSendResponse;
     property OnReceiveMessage: TMessageReceiveEvent Read FOnReceiveMessage Write FOnReceiveMessage;
     procedure CheckSynchronousSendResult(AResult: TSendResponse; AMsg: String);
 
-    // single thread
-    procedure SendMessage(AMsg: String);
+    // single thread - like SynchronousSend, but don't hold the thread waiting
+    procedure SendMessage(AMsg: AnsiString);
     // you can't call SendMessage again without calling GetReply first
-    function GetReply(var VReply: String): TSendResponse;
-    function GetMessage(var VMsg: String): TObject;  // return nil if no messages
+    function GetReply(var VReply: AnsiString): TSendResponse;
+    function GetMessage(var VMsg: AnsiString): pointer;  // return nil if no messages
     // if you don't call SendReply then no reply will be sent.
-    procedure SendReply(AMsgHnd: TObject; AReply: String);
+    procedure SendReply(AMsgHnd: pointer; AReply: AnsiString);
 
+    Function HasClientConnection : Boolean;
+    Procedure Disconnect;
   Published
     // basic properties
     property Address: String Read FAddress Write SetAddress;  // leave blank and we will be server
     property Port: Word Read FPort Write SetPort Default DEFAULT_PORT;
 
+    Property KeepAlive : Boolean read FKeepAlive write SetKeepAlive;
+
     // milliseconds - message timeout - how long we wait for other system to reply
-    property TimeOut: LongWord Read FTimeOut Write SetTimeOut Default DEFAULT_TIMEOUT;
+    Property TimeOut: Cardinal Read FTimeOut Write SetTimeOut Default DEFAULT_TIMEOUT;
 

```

Comment:
I don't see the need to change the type in these properties.

Reply:
Changed as requested
https://github.com/IndySockets/Indy/pull/442/commits/7bd8d29be4574608893391b97fcb794d959042b9